Add mig.shared.filemarks function test coverage

jonasbardino opened the pull request:
Basic `mig/shared/filemarks.py` function test coverage reusing the overall structure from the other unit tests.

jonasbardino committed:
Basic filemarks function test coverage reusing the overall structure from the
other unit tests.

jonasbardino committed:
Increase filemarks coverage with more corner cases. A few are explicitly
disabled when run as privileged user like in our doker CI because basic ACLs
are not fully enforced in that case.

albu-diku reviewed (approved):
This test is relatively strightforward. I have noted that in at least one of the tests it isn't clear which of the operations correspnding to arranging the test cirsumstances and what is the operation that is actually being checked (act), which could use some clarification.

This also has the test name and docstring duplication, as mentioned https://github.com/ucphhpc/migrid-sync/pull/390#discussion_r2657525753. It affect that pull request much more than this one, and ideally we would address that before it goes in-tree. It looks like the docstrings in this file contain a more direct description of what is being tested.

jonasbardino committed:
Address review feedback by introducing prepare and check helper methods to
clarify what is what.
Switched to static file setup and checks instead of using the API for
everything. Elaborated a bit on docs about what is tested where.

jonasbardino commented on tests/test_mig_shared_filemarks.py:
```diff
@@ -0,0 +1,221 @@
+        reset_result = reset_filemark(self.configuration, self.marks_base,
+                                      [TEST_MARKS_FILE], delete=True)
+        self.assertTrue(reset_result)
+        retrieved = get_filemark(self.configuration, self.marks_base,
```
I have updated with separate prepare and verify helpers to clarify what is what and switched them to use direct file setup and checks instead of using the API calls for that part, too.

jonasbardino commented:
> This test is relatively strightforward. I have noted that in at least one of the tests it isn't clear which of the operations correspnding to arranging the test cirsumstances and what is the operation that is actually being checked (act), which could use some clarification.
> 
> This also has the test name and docstring duplication, as mentioned [#390 (comment)](https://github.com/ucphhpc/migrid-sync/pull/390#discussion_r2657525753). It affect that pull request much more than this one, and ideally we would address that before it goes in-tree. It looks like the docstrings in this file contain a more direct description of what is being tested.

Thanks for reviewing. I've addressed the comment and a bit more.

jonasbardino committed:
Split partial success/failure care into two tests in order to switch off the
permission based one when CI runs as privileged used and always test for
another failing case where a file is in the way.

jonasbardino committed:
Remove copy/pasted left-overs in last commit.

jonasbardino commented:
Sigh, we still actually still have the CI run `make unittest` as a privileged user in GH Actions, so tests relying on strictly enforced permissions will fail there. 
Split the last test into two cases so only one depends on permissions and then decorated that one to be skipped when running as privileged user.

albu-diku commented:
> Sigh, we still actually still have the CI run `make unittest` as a privileged user in GH Actions, so tests relying on strictly enforced permissions will fail there. Split the last test into two cases so only one depends on permissions and then decorated that one to be skipped when running as privileged user.

Are GitHub actions are running as root by default!?

jonasbardino commented:
> > Sigh, we still actually still have the CI run `make unittest` as a privileged user in GH Actions, so tests relying on strictly enforced permissions will fail there. Split the last test into two cases so only one depends on permissions and then decorated that one to be skipped when running as privileged user.
> 
> Are GitHub actions are running as root by default!?

Yes, it appears so when using the container setup as we do in
https://github.com/ucphhpc/migrid-sync/blob/next/.github/workflows/ci.yml
you'll notice how we e.g. don't need to use `sudo` for `dnf` calls there.
It could be changed of course, but again requires some effort and time to get right with the cumbersome testing of Actions.
